libs/log.c

```diff
@@ -46,7 +50,31 @@ log_set_level(int ll)
 void
 log_open(void)
 {
-	char	*path = "fvwm3-output.log";
+	char	*path;
+
+	if (getenv("FVWM3_LOGFILE"))
+	{
+		path = fxstrdup(getenv("FVWM3_LOGFILE"));
+#ifdef HAVE_GETPWUID
+		if (strncmp(path, "~/", 2) == 0)
+		{
+		        /* Remove only ~ we need / between homedir and the rest */
+			memmove(path, path + 1, strlen(path));
```

Comment:
Excellent start.  This `memmove` though will overrun if path isn't NUL-terminated.  `strlen(...) + 1` will fix that.